core/src/toga/documents.py

```diff
@@ -104,10 +75,90 @@ def main_window(self) -> Window | None:
     def main_window(self, window: Window) -> None:
         self._main_window = window
 
+    @property
+    def title(self) -> str:
+        """The title of the document.
+
+        This will be used as the default title of a :any:`toga.DocumentWindow` that
+        contains the document.
+        """
+        return f"{self.document_type}: {self.path.stem if self.path else 'Untitled'}"
+
+    @property
+    def modified(self) -> bool:
+        """Has the document been modified?"""
+        return self._modified
+
+    @modified.setter
+    def modified(self, value: bool):
+        self._modified = bool(value)
+
+    ######################################################################
+    # Document operations
+    ######################################################################
+
+    def close(self):
+        """Close all the windows for this document.
+
+        This will immediately close any document windows, *without* invoking any
+        ``on_close`` handlers.
+        """
+        self.main_window.close()
```

Comment:
This doesn't seem like a very useful behavior. Toga itself calls only calls this method in response to `Document.open` throwing an exception, but since the document object would probably be in an invalid state, there's no guarantee that it would work.

Instead I think it would make more sense for `Document.close` to call `app.documents._remove`. This would provide a public interface to remove something from the DocumentSet, which the API currently lacks, and would do so in a way that's symmetrical with the document being added to the set in `Document.__init__`. 

DocumentWindow._close would then change to call `Document.close`.

Reply:
The intended use case around this was to provide a document-based entry point for closing, especially for multi-window documents to perform window cleanup. However, as you note, it's not really a useful entry point, as the usual mechanism will be to use a GUI control.

I'm not sure the reversal actually improves things much though. You'd end up with a situation where calling close on a document... doesn't actually close anything. Symmetry with add documents from the DocumentSet isn't that compelling for me either; after all, the intention of the rest of the API is that you don't have to manually add anything when you create a document, so having an API to manually *remove* seems out of place.

Honestly, I think the best option is probably to remove the API entirely, and fall back to native cleanup of the document in the once place we're actually using the API (which is in DocumentSet, so `.remove()` makes perfect sense there. If it turns out the API is needed, we can revisit that later, but all signs for now point to it being unnecessary (or, at least, requiring additional thought).